File: backend/app/src/main/java/org/example/app/controllers/requests/search/CheckAvailability.java
package org.example.app.controllers.requests.search;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Getter
@NoArgsConstructor
@SuppressWarnings("unused")
public class CheckAvailability {
    @NotNull
    private UUID courseId;
    @NotNull
    private UUID cityFromId;
    @NotNull
    private UUID cityToId;
}

File: frontend/src/app/models/request-models/checkAvailability.ts
export class CheckAvailability {
    courseId: string;
    cityFromId: string;
    cityToId: string;

    constructor(courseId: string, cityFromId: string, cityToId: string) {
        this.cityFromId = cityFromId;
        this.cityToId = cityToId;
        this.courseId = courseId;
    }
}